### DISCUSSION TITLE
When creating a user in WP GraphQL, how can I specify an avatar?

### DISCUSSION THREAD

**Discussion by maksimkoshman** [wp-graphql/wp-graphql#2914] [Q&A]
Hello! I need some help. I am creating a user, through the "registerUser" mutation. I need to be able to write the user's avatar into a field when registering. Can you tell me how to do it? 

I tried something like this, it didn't work for me:

```php
add_action('graphql_register_types', 'graphql_extension_register_types');

function graphql_extension_register_types() {
    register_graphql_field('User', 'picture', [
        'description' => __("Field for custom user's picture", 'user-avatar'),
        'type' => 'String',
    ]);
}

add_filter('graphql_input_fields', function($input_fields, $type_name) {
    if ($type_name === "RegisterUserInput") {
        $input_fields['picture'] = [
            'type' => 'String',
            'description' => __('A string containing the user\'s picture.', 'wp-graphql'),
        ];
    }

    return $input_fields;
}, 10, 2);
``` 

I really need your help

**Comment by jasonbahl** [comment]
@MaximProger it looks like this will modify the GraphQL Schema, but does not modify the resolvers to save the data being input. 

You would want to also hook into the action `graphql_user_object_mutation_update_additional_data` hook to map the input data to a function that stores the data in user meta or elsewhere.

see: 
- https://www.wpgraphql.com/actions/graphql_user_object_mutation_update_additional_data
- https://github.com/wp-graphql/wp-graphql/blob/develop/src/Data/UserMutation.php#L233
